packages/zapp/console/src/components/Executions/Tables/NodeExecutionsTable.tsx

```diff
@@ -26,16 +35,57 @@ const scrollbarPadding = scrollbarSize();
  * NodeExecutions are expandable and will potentially render a list of child
  * TaskExecutions
  */
-export const NodeExecutionsTable: React.FC<NodeExecutionsTableProps> = ({
-  abortMetadata,
-  nodeExecutions,
-}) => {
+export const NodeExecutionsTable: React.FC<NodeExecutionsTableProps> = ({ abortMetadata }) => {
   const [selectedExecution, setSelectedExecution] = React.useState<NodeExecutionIdentifier | null>(
     null,
   );
+  const [nodeExecutions, setNodeExecutions] = React.useState<NodeExecution[]>([]);
   const commonStyles = useCommonStyles();
   const tableStyles = useExecutionTableStyles();
 
+  const { compiledWorkflowClosure } = useNodeExecutionContext();
+  const { staticExecutionIdsMap } = compiledWorkflowClosure
+    ? transformerWorkflowToDag(compiledWorkflowClosure)
+    : { staticExecutionIdsMap: {} };
+  const nodeExecutionsById = React.useContext(NodeExecutionsByIdContext);
+  const dynamicParents = checkForDynamicExecutions(nodeExecutionsById, staticExecutionIdsMap);
+  const { data: dynamicWorkflows } = useQuery(
+    makeNodeExecutionDynamicWorkflowQuery(dynamicParents),
+  );
+
+  React.useEffect(() => {
+    const nodes: dNode[] = compiledWorkflowClosure
+      ? transformerWorkflowToDag(compiledWorkflowClosure, dynamicWorkflows).dag.nodes
+      : [];
+    // we remove start/end node info in the root dNode list during first assignment
+    const initializeNodes = convertToPlainNodes(nodes);
+    if (nodeExecutionsById) {
+      const executions: NodeExecution[] = [];
+      initializeNodes.map((node) => {
+        if (nodeExecutionsById[node.scopedId]) executions.push(nodeExecutionsById[node.scopedId]);
+        else
+          executions.push({
+            closure: {
+              createdAt: dateToTimestamp(new Date()),
+              outputUri: '',
+              phase: NodeExecutionPhase.UNDEFINED,
+            },
+            id: {
+              executionId: {
+                domain: node.value?.taskNode?.referenceId?.domain,
+                name: node.value?.taskNode?.referenceId?.name,
+                project: node.value?.taskNode?.referenceId?.project,
+              },
+              nodeId: node.id,
+            },
+            inputUri: '',
+            scopedId: node.scopedId,
+          });
+      });
+      setNodeExecutions(executions);
+    }
+  }, [dynamicWorkflows, compiledWorkflowClosure, nodeExecutionsById]);
```

Comment:
if we move transformer and dynamicExecutions calls to a parent level (`ExecutionTab`), this effect will go there too, leaving table a dumb render component with no complex logic